src/constructs/VPC.ts

```diff
@@ -0,0 +1,52 @@
+import { SecurityGroup, SubnetType, Vpc } from "@aws-cdk/aws-ec2";
+import { Construct as CdkConstruct } from "@aws-cdk/core";
+import { FromSchema } from "json-schema-to-ts";
+import { AwsCfInstruction } from "@serverless/typescript";
+import { AwsConstruct, AwsProvider } from "../classes";
+
+const VPC_DEFINITION = {
+    type: "object",
+    properties: {
+        type: { const: "vpc" },
+    },
+    additionalProperties: false,
+    required: [],
+} as const;
+
+type Configuration = FromSchema<typeof VPC_DEFINITION>;
+
+export class VPC extends AwsConstruct {
+    public static type = "vpc";
+    public static schema = VPC_DEFINITION;
+
+    private readonly vpc: Vpc;
+
+    constructor(scope: CdkConstruct, id: string, configuration: Configuration, private provider: AwsProvider) {
+        super(scope, id);
+
+        this.vpc = new Vpc(this, "VPC", {
+            maxAzs: 2,
+        });
+
+        const privateSubnets = this.vpc.selectSubnets({
+            subnetType: SubnetType.PRIVATE,
+        }).subnetIds;
```

Comment:
There is the `vpc.privateSubnets` property, is there any difference? (I'm just arguing for `vpc.privateSubnets` because it is simpler)

Mmh after looking up, and unless I'm mistaken, it seems to be equivalent: https://github.com/aws/aws-cdk/blob/283ed02c64f827161edba3e11c3cead3b54b7ee9/packages/%40aws-cdk/aws-ec2/lib/vpc.ts#L524-L529

Reply:
They indeed seem similar, so let's go with the more concise `vpc.privateSubnets` syntax.